### DISCUSSION TITLE
Window position

### DISCUSSION THREAD

**Discussion by pes00** [FreeRDP/FreeRDP#10571] [Q&A]
With FreeRDP 3.8 on Ubuntu Server 24.04.01LTS with LXDE. Screen resolution: 1280 x 1024.
I run the command:

` flatpak run com.freerdp.FreeRDP /u:user /p:password /v:123.123.123.123 /size:1280x900 -decorations`

The window is positioned in the center, so I have a top margin of 62px and a bottom margin of 62px. The screen looks like this:

![1](https://github.com/user-attachments/assets/5ef7c8d4-aa26-4954-9938-24c2ababfd82)

How can I position the window at the top, and have all the margin at the bottom like in this image?

![2](https://github.com/user-attachments/assets/d7601c32-2b94-4c3c-b282-17bed76d71d0)

**Comment by akallabeth** [comment]
you are most likely looking for https://github.com/jordansissel/xdotool
or you just move the window with the key combination to move a window (meta/win + mouse drag on gnome for example)

**Comment by pes00** [comment]
Thanks for your help
I'm still looking for a fix for the problem reported here:
[https://github.com/FreeRDP/FreeRDP/issues/9749](url)
(By the way... the problem is also present on Ubuntu Server 24.04.01LTS with LXDE)

Instead of using the /workarea parameter, I wanted to subtract the value of the taskbar height from the freerdp window. So if I have a resolution of 1024 (y), and the height of the bar is 30px, I thought of running freerdp with the /size:1280x994 parameter but it continues to center the window in the middle.
I honestly did not understand the solution you suggested...
I was hoping that the /window-position parameter could be useful, but I do not understand how to use it. When I use it nothing changes.

**Comment by pes00** [comment]
I studied the solution with xdotool but it doesn't work with wayland and it seems there is no alternative.
Is it possible that there is no solution that allows you to have the freerdp window and the operating system bar? In version 2 of freerdp the workarea function worked perfectly

**Reply by akallabeth** [reply]
@pes00 that was not your question, was it?
also, just tested what you said, both, `xfreerdp` version 2 and 3 as well as `sdl-freerdp` behave completely the same way with `xfreerdp /v:host -decorations /workarea` (that is, `debian gnome wayland`)

so, please be more precise: what are you actually asking?

**Comment by pes00** [comment]
akallabeth, I've been looking for a solution to this problem for months.
I originally opened this post:
https://github.com/FreeRDP/FreeRDP/issues/9749

but the issue was closed without a solution. I tried the /workarea option on both Ubuntu and Raspberry OS and both have the same problem.
So I tried to change my approach, and not use /workarea. I then opened this thread, but the suggestion to use xdotool does not work on wayland, so I am back to square one.
What I would like is the FreeRDP 3 window and the OS bar together, without overlapping. Basically what /workarea should do.
I have done dozens of tests but there is no solution.
I assure you that the /workarea option on Ubuntu and Raspberry OS does not work correctly. You have to believe me.

**Reply by akallabeth** [reply]
@pes00 ok, that explains a lot :grin: 
sorry, but I can understand your fustration, but I did not even remember that issue (there are lots of similar requests, mostly boiling down to some configuration quirks)

might help if you narrow down to what exactly triggers that issue for you (and describe the configurations  you tested a little more)
that `xfreerdp` does not work is not really something that you mentioned in the referenced issue.

**Comment by pes00** [comment]
As for the idea of ​​moving the window, I think there is nothing else to say, or am I wrong? On Wayland it seems there is no possibility.
For the problem with /workarea, I can detail the tests done on Ubuntu
Do you want me to continue in this discussion or do you prefer that I reopen the previous post by putting the /workarea problem also with Ubuntu?

**Reply by akallabeth** [reply]
@pes00 well, depends on which client you use (and if it is `X11` or `Wayland`)
did some tests here too:
* debian / gnome / wayland
* `sdl-freerdp /v:host -decorations /workarea` does perfectly fit the usable desktop area. (if that does not work in your case -> report to https://github.com/libsdl-org/SDL/ )
* `xfreerdp` did have an issue fixed with https://github.com/FreeRDP/FreeRDP/pull/10671 (that is `_NET_CURRENT_DESKTOP` does not work with certain window managers)

**Comment by pes00** [comment]
I didn't know sdl-freerdp...
Is it possible to use it with flatpak? If so, what is the command?

**Reply by akallabeth** [reply]
@pes00 `sdl-freerdp` is the default client launched in the flatpak.
I´ve given you the command to start `xfreerdp` back then `flatpak run --command=xfreerdp com.freerdp.FreeRDP ...`

**Comment by pes00** [comment]
Ok, this is the command I used in all the tests done with FreeRDP 3.8
`flatpak run com.freerdp.FreeRDP /u:user /p:password /v:111.111.111.111 /workarea -decorations /cert:ignore /network:auto`
and these are the results.

Raspberry pi OS Bookworm 64bit - Wayfire window manager with Wayland backend:
![Raspberry_bookwarm_Wayland](https://github.com/user-attachments/assets/d115e869-b414-479f-9fe8-3a127dd017ab)

Raspberry pi OS Bookworm 64bit - Openbox window manager with X11 backend:
![Raspberry_bookwarm_X11](https://github.com/user-attachments/assets/469a2420-1b5a-4162-99ce-6a682725f581)

Ubuntu Server 24.04.01 LTS 64bit with LXDE
![Ubuntu_LXDE](https://github.com/user-attachments/assets/36ee0f3b-1d8d-4e10-a738-21c80d003283)

**Comment by pes00** [comment]
@akallabeth 
Are the tests I have done sufficient or do you need more?
Do you think the problem can be solved?

**Reply by akallabeth** [reply]
@pes00 did not yet have time to check up on this, will take a little time, sorry

**Reply by pes00** [reply]
I installed version 3.9 but unfortunately the problem is still there.
I just ask if you still need time to address this problem or if it has been abandoned altogether

**Reply by akallabeth** [reply]
@pes00 no, it was just not important enough to delay 3.9.0

**Comment by pes00** [comment]
Absolutely, no problem. I just wanted to know if you needed more information.
Thanks for the work you are doing

**Comment by akallabeth** [comment]
moved to an issue to keep track of what is going on.
